packages/backend/src/open_payments/auth/middleware.ts

```diff
@@ -2,6 +2,87 @@ import { AccessType, AccessAction } from './grant'
 import { PaymentPointerContext } from '../../app'
 import { Transaction } from 'objection'
 import { GrantReference } from '../grantReference/model'
+import { createVerifier, httpis, RequestLike } from 'http-message-signatures'
+import { KeyObject } from 'crypto'
+import { Request as KoaRequest } from 'koa'
+import { ClientKeys } from '../../clientKeys/model'
+import { JWKWithRequired } from 'auth'
+
+// Creates a RequestLike object for the http-message-signatures library input
+function requestLike(request: KoaRequest): RequestLike {
+  return {
+    method: request.method,
+    headers: request.headers,
+    url: request.url
+  }
+}
+
+function parseJwkKeyType(jwk: JWKWithRequired): KeyType {
+  if (jwk.kty === 'private' || jwk.kty === 'public' || jwk.kty === 'secret') {
+    return jwk.kty
+  } else {
+    throw new Error('invalid key type')
+  }
+}
+
+function parseJwkUsages(jwk: JWKWithRequired): Array<KeyUsage> {
+  const { use } = jwk
+  if (
+    use === 'decrypt' ||
+    use === 'deriveBits' ||
+    use === 'deriveKey' ||
+    use === 'encrypt' ||
+    use === 'sign' ||
+    use === 'unwrapKey' ||
+    use === 'verify' ||
+    use === 'wrapKey'
+  ) {
+    return [use]
+  } else if (use === undefined) {
+    return []
+  } else {
+    throw new Error('invalid usage')
```

Comment:
:confused:  The OpenAPI spec appears to only allow `sig`
https://github.com/interledger/open-payments/blob/4c873dba89164decffbe84905d12f1d4ec045389/auth-server-open-api-spec.yaml#L677-L680